homeassistant/components/mqtt/__init__.py

```diff
@@ -351,17 +331,29 @@ def async_setup(hass, config):
         return False
 
     # For cloudmqtt.com, secured connection, auto fill in certificate
-    if certificate is None and 19999 < port < 30000 and \
-       broker.endswith('.cloudmqtt.com'):
+    if (certificate is None and 19999 < port < 30000 and
+            broker.endswith('.cloudmqtt.com')):
         certificate = os.path.join(os.path.dirname(__file__),
                                    'addtrustexternalcaroot.crt')
 
     # When the certificate is set to auto, use bundled certs from requests
     if certificate == 'auto':
         certificate = requests.certs.where()
 
-    will_message = conf.get(CONF_WILL_MESSAGE)
-    birth_message = conf.get(CONF_BIRTH_MESSAGE)
+    will_message = None
+    if conf.get(CONF_WILL_MESSAGE) is not None:
+        will_conf = conf.get(CONF_WILL_MESSAGE)
+        will_message = Message(will_conf.get(ATTR_TOPIC),
+                               will_conf.get(ATTR_PAYLOAD),
+                               will_conf.get(ATTR_QOS),
+                               will_conf.get(ATTR_RETAIN))
```

Comment:
The `Message` namedtuple also has the added benefit of not having to pass around configuration objects - please tell me if this is not wanted/should be in another PR.

Reply:
Fine to include in this PR.

Reply:
```python
will_message = Message(**will_conf)
```

Reply:
Oh, that's very nice indeed!